url/src/origin.rs

```diff
@@ -93,7 +92,11 @@ impl Origin {
             Origin::Tuple(ref scheme, ref host, port) => {
                 let host = match *host {
                     Host::Domain(ref domain) => {
-                        let (domain, _errors) = domain_to_unicode(domain);
+                        #[cfg(feature = "idna")]
+                        let (domain, _errors) = idna::domain_to_unicode(domain);
+                        #[cfg(not(feature = "idna"))]
+                        let domain = domain.clone();
```

Comment:
Like with the above comment, this should maybe panic if `domain` would change after `idna::domain_to_unicode` (if we can determine that easily?); otherwise, this function would return two different things based on whether the feature is enabled or not.

Reply:
If idna feature disabled, then the whole fn unicode_serialization can be opt out? I searched spec about unicode_serialization, it is strictly related with idna.

Reply:
https://html.spec.whatwg.org/multipage/origin.html#unicode-serialisation-of-an-origin now the spec even deletes the section of unicode serialization, just saying "There used to also be a Unicode serialization of an origin. However, it was never widely adopted."